client.go

```diff
@@ -1375,3 +1398,455 @@ func (c *ToznySDKV3) UnbrokerShare(ctx context.Context, authorizerClientID strin
+	var sharedSecrets ListedSecrets
+
+	listRequest := storageClient.ListGroupsRequest{
+		ClientID:  uuid.MustParse(c.StorageClient.ClientID),
+		NextToken: nextToken,
+		Max:       limit,
+	}
+	responseList, err := c.StorageClient.ListGroups(ctx, listRequest)
+	if err != nil {
+		return sharedSecrets, err
+	}
+	if len(responseList.Groups) < 1 {
+		return sharedSecrets, nil
+	}
+	var found bool
+	sharedSecretList := []pdsClient.ListedRecord{}
+	for index, s := range responseList.Groups {
+		listReq := storageClient.ListGroupRecordsRequest{
+			GroupID: s.GroupID,
+		}
+		listGroupRecords, err := c.StorageClient.GetSharedWithGroup(ctx, listReq)
+		if err != nil {
+			return sharedSecrets, err
+		}
+		if len(listGroupRecords.ResultList) >= 1 {
+			for _, r := range listGroupRecords.ResultList {
+				found = false
+				for _, l := range sharedSecretList {
+					if r.Metadata.RecordID == l.Metadata.RecordID {
+						found = true
+					}
+				}
+				if !found {
+					shared := "No"
+					username := ""
+					if responseList.Groups[index].MemberCount > 1 {
+						shared = "Yes"
+					}
+					// find the username for secret writer if it's someone else
+					searchParams := identityClient.SearchRealmIdentitiesRequest{
+						RealmName:         login.RealmName,
+						IdentityClientIDs: []uuid.UUID{uuid.MustParse(r.Metadata.WriterID)},
+					}
+					identities, err := c.E3dbIdentityClient.SearchRealmIdentities(ctx, searchParams)
+					if err != nil {
+						return sharedSecrets, err
+					}
+					username = identities.SearchedIdentitiesInformation[0].RealmUsername
+					r.Metadata.Plain["username"] = username
+					r.Metadata.Plain["shared"] = shared
+					r, err = c.DecryptFromListedRecord(ctx, r)
+					if err != nil {
+						return sharedSecrets, err
+					}
+					sharedSecretList = append(sharedSecretList, r)
+				}
+			}
+		}
+	}
+	sharedSecrets.List = sharedSecretList
+	sharedSecrets.NextToken = fmt.Sprintf("%d", responseList.NextToken)
+	return sharedSecrets, nil
+}
+
+func (c *ToznySDKV3) ViewSecret(ctx context.Context, secretID string) (pdsClient.ListedRecord, error) {
+	// var secret pdsClient.ListedRecord
+	secret := pdsClient.ListedRecord{}
+	listRequest := storageClient.ListGroupsRequest{
+		ClientID: uuid.MustParse(c.StorageClient.ClientID),
+	}
+	groupList, err := c.StorageClient.ListGroups(ctx, listRequest)
+	if err != nil {
+		return secret, err
+	}
+	for _, s := range groupList.Groups {
+		listReq := storageClient.ListGroupRecordsRequest{
+			GroupID: s.GroupID,
+		}
+		listGroupRecords, err := c.StorageClient.GetSharedWithGroup(ctx, listReq)
+		if err != nil {
+			return secret, err
+		}
+		if len(listGroupRecords.ResultList) >= 1 {
+			for _, record := range listGroupRecords.ResultList {
+				if record.Metadata.RecordID == secretID {
+					secret = record
+				}
+			}
+		}
+	}
+	// otherwise get secret from readRecord
```

Comment:
talk to @rosasck @lkwdwrd , this hack should no longer be needed

Reply:
@carrala is this pulling a record thats not a secret?

Reply:
@rosasck it's getting the secrets if it's not in any of the groups. but levi is correct that it's a bit redundant since even secrets owned by the reader are in at least one group, so the record will definitely be discovered by the for loop above.